File: server/RestApiServer/Dto/Admin/AdminPortalAppState.cs
using RestApiServer.Db.Users;
using RestApiServer.Dto.App;

namespace RestApiServer.Dto.Admin
{
    public class AdminPortalAppState
    {
        public required LoggedInUserInfo CurrentLoggedInUser { get; set; }
        public required AdminPortalStats AdminPortalStats { get; set; }
    }
}

File: adminportal/src/Dto/AdminPortal/AdminPortalAppState.ts
import type { LoggedInUserInfo } from "../LoggedInUserInfo"

export type AdminPortalAppState = {
    currentLoggedInUser: LoggedInUserInfo | null,
    adminPortalStats: AdminPortalStats
}

export type AdminPortalStats = {
    totalCategories: number;
    totalBoards: number;
    totalUsers: number;
    totalThreads: number;
    totalTopics: number;
    totalPosts: number;
    totalGalleryItems: number;
    totalImages: number;
    totalVideos: number;
    totalFiles: number;
    totalAudioFiles: number;
}